worlds/AutoWorld.py

```diff
@@ -388,6 +388,19 @@ def collect_item(self, state: "CollectionState", item: "Item", remove: bool = Fa
     def get_pre_fill_items(self) -> List["Item"]:
         return []
 
+    def filter_early_locations(self, loc_list: Sequence[Location]) -> Sequence[bool]:
+        """
+        `loc_list` is a sequence (`list`) of `Location`s for your game
+        that our algorithms have guessed are "early" locations.
```

Comment:
I don't see the value of putting information about how the internals are working in the documentation of the API.
The API should only be what someone using the API needs to care about.
There is most certainly a guessing algorithm involved. We can see the incorrect guesses, which is the reason this is a problem.
If we decide to change the strategy on guessing which locations are early, that would still be guessing, and we shouldn't need to change the API because of that.